### PR TITLE
Rename common mounts to auxiliary images

### PR DESCRIPTION
**Pull request by rjeberhard**
During development we have called this new feature "common mounts." Prior to releasing the feature we are updating the name of the feature to "auxiliary images." 

This affects schema elements, documentation, samples, and the implementation.

All ItMii* integration tests are passing: https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/5604/
Full integration test job is running now.

### CODE REVIEW

**Review by ddsharpe — APPROVED** [review]

File: operator/integration-tests/model-in-image/run-test.sh [hunk 1]
```diff
@@ -75,7 +75,7 @@ function usage() {
                 Note that this depends on the db being deployed
                 and initialized via rcu. So either pre-deploy
                 the db and run rcu or pass '-db' and '-rcu' too.
-    -cm       : Run the tests in common mounts mode.
+    -aux      : Run the tests in auxiliary images mode.
```
Comment (tbarnes-us) [inline operator/integration-tests/model-in-image/run-test.sh:78]: `-aux` should be `-ai` per the command line argument processing below

Comment (rjeberhard) [reply operator/integration-tests/model-in-image/run-test.sh:78]: Looks like it... Thanks.

**Comment by tbarnes-us** [comment]
General comment:

The output of  `kubectl explain domain.spec.serverPod.auxiliaryImages` and `kubectl explain domain.spec.auxiliaryImageVolumes` both have a bug -- the 'DESCRIPTION' paragraph of each is doubled. This is not a regression from the rename -- I very recently noticed this error was occurring before the rename -- but I'm hoping you know of a simple fix off the top of your head.

File: operator/src/main/java/oracle/kubernetes/operator/helpers/BasePodStepContext.java [hunk 2]
```diff
@@ -51,32 +51,32 @@
...
 
-  private boolean hasMatchingVolumeName(CommonMountVolume commonMountVolume, CommonMount commonMount) {
-    return commonMount.getVolume().equals(commonMountVolume.getName());
+  private boolean hasMatchingVolumeName(AuxiliaryImageVolume auxiliaryImageVolume, AuxiliaryImage auxiliaryImage) {
+    return auxiliaryImage.getVolume().equals(auxiliaryImageVolume.getName());
   }
 
-  protected void addVolumeMount(V1Container container, CommonMount cm) {
-    Optional.ofNullable(getMountPath(cm, info.getDomain().getCommonMountVolumes())).ifPresent(mountPath ->
+  protected void addVolumeMount(V1Container container, AuxiliaryImage cm) {
```
Comment (ankedia) [inline operator/src/main/java/oracle/kubernetes/operator/helpers/BasePodStepContext.java:64]: Minor - parameter name `cm` can be changes to `ai`.

File: operator/src/main/java/oracle/kubernetes/operator/helpers/BasePodStepContext.java [hunk 3]
```diff
@@ -51,32 +51,32 @@
...
+    Optional.ofNullable(getMountPath(cm, info.getDomain().getAuxiliaryImageVolumes())).ifPresent(mountPath ->
             addVolumeMountIfMissing(container, cm, mountPath));
   }
 
-  protected void addVolumeMountIfMissing(V1Container container, CommonMount cm, String mountPath) {
+  protected void addVolumeMountIfMissing(V1Container container, AuxiliaryImage cm, String mountPath) {
```
Comment (ankedia) [inline operator/src/main/java/oracle/kubernetes/operator/helpers/BasePodStepContext.java:69]: Minor - parameter name `cm` can be changes to `ai`.

File: operator/src/main/java/oracle/kubernetes/operator/helpers/BasePodStepContext.java [hunk 4]
```diff
@@ -96,58 +96,59 @@ protected V1Container createPrimaryContainer(TuningParameters tuningParameters)
         .securityContext(getServerSpec().getContainerSecurityContext());
   }
 
-  protected V1Volume createEmptyDirVolume(CommonMountVolume cmv) {
+  protected V1Volume createEmptyDirVolume(AuxiliaryImageVolume cmv) {
```
Comment (ankedia) [inline operator/src/main/java/oracle/kubernetes/operator/helpers/BasePodStepContext.java:99]: Minor - parameter name `cmv` can be changes to `aiv`.

File: operator/src/test/java/oracle/kubernetes/operator/helpers/PodHelperTestBase.java [hunk 5]
```diff
@@ -678,125 +677,129 @@ public void whenPodCreated_withNoPvc_fromModel_containerHasExpectedVolumeMounts(
...
 
     assertThat(getCreatedPod().getSpec().getVolumes(),
-            hasItem(new V1Volume().name(getCommonMountVolumeName()).emptyDir(
+            hasItem(new V1Volume().name(getAuxiliaryImageVolumeName()).emptyDir(
                     new V1EmptyDirVolumeSource().sizeLimit(Quantity.fromString("100G")))));
   }
 
   @Test
-  public void whenDomainHasCommonMountsWithImagePullPolicy_createPodsWithCMInitContainerHavingImagePullPolicy() {
+  public void whenDomainHasAuxiliaryImagesWithImagePullPolicy_createPodsWithCMInitContainerHavingImagePullPolicy() {
```
Comment (ankedia) [inline operator/src/test/java/oracle/kubernetes/operator/helpers/PodHelperTestBase.java:741]: Minor - method name can be changed to `whenDomainHasAuxiliaryImagesWithImagePullPolicy_createPodsWithAusInitContainerHavingImagePullPolicy`

File: operator/src/test/java/oracle/kubernetes/weblogic/domain/model/DomainValidationTest.java [hunk 6]
```diff
@@ -112,78 +112,78 @@ public void whenManagerServerSpecsHaveDns1123DuplicateNames_reportError() {
   }
 
   @Test
-  public void whenDomainConfiguredWithCommonMountButNoCommomMountVolumes_reportError() {
+  public void whenDomainConfiguredWithAuxiliaryImageButNoCommomMountVolumes_reportError() {
```
Comment (ankedia) [inline operator/src/test/java/oracle/kubernetes/weblogic/domain/model/DomainValidationTest.java:115]: Minor - test method name - `whenDomainConfiguredWithAuxiliaryImageButNoAuxImageVolumes_reportError`

File: operator/src/test/java/oracle/kubernetes/weblogic/domain/model/DomainValidationTest.java [hunk 7]
```diff
@@ -112,78 +112,78 @@ public void whenManagerServerSpecsHaveDns1123DuplicateNames_reportError() {
...
 
     assertThat(domain.getValidationFailures(resourceLookup),
-            contains(stringContainsInOrder("serverPod.commonMounts",
+            contains(stringContainsInOrder("serverPod.auxiliaryImages",
                     "there is no matching volume defined with name 'test'")));
   }
 
   @Test
-  public void whenDomainConfiguredWithCommonMountButNoMatchingCommomMountVolumes_reportError() {
+  public void whenDomainConfiguredWithAuxiliaryImageButNoMatchingCommomMountVolumes_reportError() {
```
Comment (ankedia) [inline operator/src/test/java/oracle/kubernetes/weblogic/domain/model/DomainValidationTest.java:125]: Minor - test method name ->
whenDomainConfiguredWithAuxiliaryImageButNoMatchingAuxImageVolumes_reportError

File: operator/src/test/java/oracle/kubernetes/weblogic/domain/model/DomainValidationTest.java [hunk 8]
```diff
@@ -112,78 +112,78 @@ public void whenManagerServerSpecsHaveDns1123DuplicateNames_reportError() {
...
+            .withAuxiliaryImages(Collections.singletonList(new AuxiliaryImage().image("wdt-image:v1")));
 
     assertThat(domain.getValidationFailures(resourceLookup),
-            contains(stringContainsInOrder("serverPod.commonMounts", "does not define a volume name")));
+            contains(stringContainsInOrder("serverPod.auxiliaryImages", "does not define a volume name")));
   }
 
   @Test
-  public void whenDomainConfiguredWithCommonMountAndMatchingCommomMountVolumes_noErrorsReported() {
+  public void whenDomainConfiguredWithAuxiliaryImageAndMatchingCommomMountVolumes_noErrorsReported() {
```
Comment (ankedia) [inline operator/src/test/java/oracle/kubernetes/weblogic/domain/model/DomainValidationTest.java:146]: Minor - test method name-> whenDomainConfiguredWithAuxiliaryImageAndMatchingAuxImageVolumes_noErrorsReported

**Review by ankedia — APPROVED** [review]
LGTM. There are few minor comments (probably missed due to typo in common mount).

File: operator/integration-tests/model-in-image/mii-sample-wrapper/README [hunk 9]
```diff
@@ -62,7 +62,7 @@ IMAGE_PULL_SECRET_NAME ::: Image pull secret name.
 
 IMAGE_TYPE ::: WDT image type.
   Defaults to $WDT_DOMAIN_TYPE for a standard WLS domain, 'RestrictedJRF', or 'JRF.
-  For the common mounts mode, this is set to $WDT_DOMAIN_TYPE-CM.
+  For the auxiliary image mode, this is set to $WDT_DOMAIN_TYPE-AUX.
```
Comment (tbarnes-us) [inline operator/integration-tests/model-in-image/mii-sample-wrapper/README:65]: AUX --> AI (and any other ocurrances of aux/AUX)

**Review by tbarnes-us — APPROVED** [review]
LGTM apart from my two remaining minor comments. I rebuilt and ran hugo plus the IT sample stand-alone to double-check.

**Pull request merged**